docs/docs/visual-testing-with-storybook.md

```diff
@@ -21,6 +21,130 @@ npx -p @storybook/cli sb init
+    config.module.rules[0].exclude = [/node_modules\/(?!(gatsby)\/)/]
+
+    // use installed babel-loader which is v8.0-beta (which is meant to work with @babel/core@7)
+    config.module.rules[0].use[0].loader = require.resolve("babel-loader")
+
+    // use @babel/preset-react for JSX and env (instead of staged presets)
+    config.module.rules[0].use[0].options.presets = [
+      require.resolve("@babel/preset-react"),
+      require.resolve("@babel/preset-env"),
+    ]
+
+    config.module.rules[0].use[0].options.plugins = [
+      // use @babel/plugin-proposal-class-properties for class arrow functions
+      require.resolve("@babel/plugin-proposal-class-properties"),
+      // use babel-plugin-remove-graphql-queries to remove static queries from components when rendering in storybook
+      require.resolve("babel-plugin-remove-graphql-queries"),
+    ]
+
+    // Prefer Gatsby ES6 entrypoint (module) over commonjs (main) entrypoint
+    config.resolve.mainFields = ["browser", "module", "main"]
+
+    return config
+  },
+  // highlight-end
+}
+```
+
+Next create a new file under `.storybook` called `preview.js`. This configuration file preview.js is not responsible for loading any stories. Its main purpose is to add global parameters and decorators.
+
+```js:title=.storybook/preview.js
+import { action } from "@storybook/addon-actions"
+
+// automatically import all files ending in *.stories.js
+// configure(require.context("../src", true, /\.stories\.js$/), module)
+
+// Gatsby's Link overrides:
+// Gatsby defines a global called ___loader to prevent its method calls from creating console errors you override it here
+global.___loader = {
+  enqueue: () => {},
+  hovering: () => {},
+}
+// Gatsby internal mocking to prevent unnecessary errors in storybook testing environment
+global.__PATH_PREFIX__ = ""
+// This is to utilized to override the window.___navigate method Gatsby defines and uses to report what path a Link would be taking us to if it wasn't inside a storybook
+window.___navigate = pathname => {
+  action("NavigateTo:")(pathname)
+}
+```
+
+### Add TypeScript Support
+
+To configure TypeScript with Storybook and Gatsby, add the following configuration to `.storybook/main.js` after line 25 containing `config.resolve.mainFields = ["browser", "module", "main"];` and before line 27 containing `return config;`
+
+```js:title=.storybook/main.js
+config.module.rules.push({
+  test: /\.(ts|tsx)$/,
+  loader: require.resolve("babel-loader"),
+  options: {
+    presets: [["react-app", { flow: false, typescript: true }]],
+    plugins: [
+      require.resolve("@babel/plugin-proposal-class-properties"),
+      // use babel-plugin-remove-graphql-queries to remove static queries from components when rendering in storybook
+      require.resolve("babel-plugin-remove-graphql-queries"),
+    ],
+  },
+})
+
+config.resolve.extensions.push(".ts", ".tsx")
+```
+
+## Storybook versions 5.0, 5.1, and 5.2
```

Comment:
@aaronsturm if you can remove this portion it would be good. On a recent conversation with one of the Storybook maintainers, he brought to my attention that they're pushing for version >=5.3 as much as possible, to provide early adoption as much as possible. With that in mind i coordinated with him to update some of the tutorials to address this push. Now when a reader  sees this document and follows it, he/she will be using the latest version with all it's features. If by any chance the reader decides to go and lookup the starter library and doing a quick check on them, it seems that only one is currently updated. more specifically [this](https://github.com/georgewilde/gatsby-starter-wilde-creations), the other ones available haven't been updated from 3 months to almost a year. So with that in mind the reader should be prompted to direct any issues to the respective repos and bring to the attention of their maintainers that he has some issues running any of the starters and with that probably avoid creating issues in the Gatsby repo that could be not related. Sounds good? Also @laurieontech are you on board with this?

Reply:
Regarding 4.0 section. It should be removed as there's alot of breaking changes between 4.x and 5.2 and from 5.2 to 5.3. When the reader follows the documentation he/she will be using the latest release of the package.

Regarding 5.2 i would leave it in but with a word of advice regarding the starters as most of them are not currently updated. Like i mentioned above.